### PR TITLE
Prefill quick open

### PR DESCRIPTION
**Pull request by Michel73**
Implements #37932

### CODE REVIEW

**Review by ctrlaltvikas — APPROVED** [review]

File: src/vs/base/parts/quickopen/browser/quickOpenWidget.ts [hunk 1]
```diff
@@ -596,8 +600,13 @@ export class QuickOpenWidget extends Disposable implements IModelProvider {
 
 		// Show based on param
 		if (types.isString(param)) {
+			this.withPrefix = true;
 			this.doShowWithPrefix(param);
 		} else {
+			this.withPrefix = false;
+			if (this.options.prefill) {
```
Comment (chrmarti) [inline src/vs/base/parts/quickopen/browser/quickOpenWidget.ts]: Could you instead pass the last value as `options.value` to show()? That will keep the widget separated from the logic in `quickOpenController.ts`.

File: src/vs/base/parts/quickopen/browser/quickOpenWidget.ts [hunk 2]
```diff
@@ -793,6 +810,10 @@ export class QuickOpenWidget extends Disposable implements IModelProvider {
 		this.builder.hide();
 		this.builder.domBlur();
 
+		// In case of quick search store entered value
+		if (!this.withPrefix) {
```
Comment (chrmarti) [inline src/vs/base/parts/quickopen/browser/quickOpenWidget.ts]: Note that one can start with a prefix, but then delete it and the other way around.

File: src/vs/workbench/browser/parts/quickopen/quickOpenController.ts [hunk 3]
```diff
@@ -133,6 +133,10 @@ export class QuickOpenController extends Component implements IQuickOpenService
 		} else {
 			this.closeOnFocusLost = this.configurationService.getValue(CLOSE_ON_FOCUS_LOST_CONFIG);
 		}
+		if (this.quickOpenWidget) {
```
Comment (chrmarti) [inline src/vs/workbench/browser/parts/quickopen/quickOpenController.ts]: Is this fixing a bug? (Or why is it now needed?)

**Review by chrmarti — CHANGES_REQUESTED** [review]

File: src/vs/workbench/browser/parts/quickopen/quickOpenController.ts (hunk 3 above)
Comment (Michel73) [reply src/vs/workbench/browser/parts/quickopen/quickOpenController.ts]: I dispose and set to null to enforce  a new creation of the quickOpenWidget to get it loaded with the changed configuration.

File: src/vs/base/parts/quickopen/browser/quickOpenWidget.ts (hunk 1 above)
Comment (Michel73) [reply src/vs/base/parts/quickopen/browser/quickOpenWidget.ts]: But how can I get the `lastInputValue` into the `quickOpenController`?

File: src/vs/base/parts/quickopen/browser/quickOpenWidget.ts (hunk 2 above)
Comment (Michel73) [reply src/vs/base/parts/quickopen/browser/quickOpenWidget.ts]: I I've considered that now.

File: src/vs/workbench/browser/parts/quickopen/quickOpenController.ts (hunk 3 above)
Comment (Michel73) [reply src/vs/workbench/browser/parts/quickopen/quickOpenController.ts]: See my last PR-commit.

**Comment by Michel73** [comment]
I had some problems synchronising with master repo. I had to rebase my branch. Now I did all the changes within a commit.

**Pull request merged**